src/main/java/org/shanerx/tradeshop/commands/EditCommand.java

```diff
@@ -281,123 +282,96 @@ private StaticGuiElement shopItemEditMenu(int index, boolean isCost) {
             ));
 
             //Add new item settings below
-            try {
-                if (item.getItemStack().getItemMeta() instanceof Damageable) {
-                    itemGroup.addElement(numericalOption("CompareDurability", tempItem, new ItemStack[]{
-                            FALSE_ITEM, new ItemStack(Material.IRON_BLOCK), TRUE_ITEM, new ItemStack(Material.GOLD_BLOCK)
-                    }));
-                }
+            if (item.getItemStack().getItemMeta() instanceof Damageable) {
+                itemGroup.addElement(numericalOption(ShopItemStackSettingKeys.compareDurability, tempItem, new ItemStack[]{
+                        FALSE_ITEM, new ItemStack(Material.IRON_BLOCK), TRUE_ITEM, new ItemStack(Material.GOLD_BLOCK)
+                }));
+            }
 
-                if (tempItem.getItemStack().getItemMeta() instanceof BookMeta) {
-                    itemGroup.addElement(booleanOption("CompareBookAuthor", tempItem, TRUE_ITEM, FALSE_ITEM));
-                    itemGroup.addElement(booleanOption("CompareBookPages", tempItem, TRUE_ITEM, FALSE_ITEM));
-                }
+            if (tempItem.getItemStack().getItemMeta() instanceof BookMeta) {
+                itemGroup.addElement(booleanOption(ShopItemStackSettingKeys.compareBookAuthor, tempItem, TRUE_ITEM, FALSE_ITEM));
+                itemGroup.addElement(booleanOption(ShopItemStackSettingKeys.compareBookPages, tempItem, TRUE_ITEM, FALSE_ITEM));
+            }
 
-                if (tempItem.getItemStack().getType().toString().endsWith("SHULKER_BOX")) {
-                    itemGroup.addElement(booleanOption("CompareShulkerInventory", tempItem, TRUE_ITEM, FALSE_ITEM));
-                }
+            if (tempItem.getItemStack().getItemMeta() instanceof FireworkMeta) {
+                itemGroup.addElement(booleanOption(ShopItemStackSettingKeys.compareFireworkDuration, tempItem, TRUE_ITEM, FALSE_ITEM));
+                itemGroup.addElement(booleanOption(ShopItemStackSettingKeys.compareFireworkEffects, tempItem, TRUE_ITEM, FALSE_ITEM));
+            }
 
-                itemGroup.addElement(booleanOption("CompareName", tempItem, TRUE_ITEM, FALSE_ITEM));
-                itemGroup.addElement(booleanOption("CompareLore", tempItem, TRUE_ITEM, FALSE_ITEM));
-                itemGroup.addElement(booleanOption("CompareUnbreakable", tempItem, TRUE_ITEM, FALSE_ITEM));
-                itemGroup.addElement(booleanOption("CompareEnchantments", tempItem, TRUE_ITEM, FALSE_ITEM));
-                itemGroup.addElement(booleanOption("CompareItemFlags", tempItem, TRUE_ITEM, FALSE_ITEM));
-                itemGroup.addElement(booleanOption("CompareCustomModelData", tempItem, TRUE_ITEM, FALSE_ITEM));
-                itemGroup.addElement(booleanOption("CompareAttributeModifier", tempItem, TRUE_ITEM, FALSE_ITEM));
```

Comment:
what happened here exactly?

Reply:
I got rid of all the separate methods for this and replaced with a map<Enum key, ObjectHolder> so they are easier to retrieve and add onto later when new setting(like the fireworks) need to be added. These methods changed from their strings to the Enum values, although I didn't full caps them